pkg/types/baremetal/validation/platform.go

```diff
@@ -360,6 +360,21 @@ func validateBootMode(hosts []*baremetal.Host, fldPath *field.Path) (errors fiel
 	return
 }
 
+// validateProvisioningNetworkDisabledSupported validates hosts bmc address support provisioning network is disabled
+func validateProvisioningNetworkDisabledSupported(hosts []*baremetal.Host, fldPath *field.Path) (errors field.ErrorList) {
+	for idx, host := range hosts {
+		accessDetails, err := bmc.NewAccessDetails(host.BMC.Address, host.BMC.DisableCertificateVerification)
+		if err != nil {
+			errors = append(errors, field.Invalid(fldPath.Index(idx).Child("BMC"), host.BMC.Address, err.Error()))
+		} else if accessDetails.RequiresProvisioningNetwork() {
```

Comment:
minor: for the happy path I guess there could be an immediate return in case of error (and else removed)?

Reply:
Yep, good point